services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -710,8 +717,638 @@ func TestCreateApplicationVersion(t *testing.T) {
 			if err == nil {
 				t.Fatalf("expected error, got none.")
 			}
-			if err.Error() != tc.expectedErrorMsg {
-				t.Fatalf("Expected different error (expected: %s, got: %s)", tc.expectedErrorMsg, err.Error())
+			var expectedErr api.CreateReleaseResponse
+			if err := prototext.Unmarshal([]byte(tc.expectedErrorMsg), &expectedErr); err != nil {
+				t.Fatalf("failed to unmarshal the expected error object: %v", err)
+			}
+			
+			var actualErr api.CreateReleaseResponse
+			if err := prototext.Unmarshal([]byte(err.Error()), &actualErr); err != nil {
+				t.Fatalf("failed to unmarshal the actual error object: %v", err)
+			}
+
+			if !proto.Equal(&expectedErr, &actualErr) {
+				t.Fatalf("Expected different error (expected: %v, got: %v)", expectedErr, actualErr)
+			}
+		})
+	}
+}
+
+func dumpFilesystemHelper(fs billy.Filesystem, path string) []string {
```

Comment:
please give it a nicer name, for example:
```suggestion
func listFilesRecursivel(fs billy.Filesystem, path string) []string {
```